File: js/problem003.js
const { isPrime } = require("./isPrime");

function solution() {
  let factors = [];
  const n = 600851475143;
  let current = n;
  if (n % 3 == 0) {
    factors.push(3);
    current /= 3;
  }
  const root = Math.floor(Math.sqrt(n));
  let maybePrime = 5;
  while (maybePrime < Math.floor(Math.sqrt(current))) {
    if (isPrime(maybePrime) && n % maybePrime == 0) {
      factors.push(maybePrime);
      current /= maybePrime;
    }
    maybePrime += 2;
    if (maybePrime <= root && isPrime(maybePrime) && n % maybePrime == 0) {
      factors.push(maybePrime);
      current /= maybePrime;
    }
    maybePrime += 4;
  }
  factors.push(current);
  return factors;
}

console.log(solution());

File: dart/problem003.dart
import 'dart:math';
import './isPrime.dart';

List<int> solution(int n) {
  var root = sqrt(n).floor();
  List<int> factors = [];
  var current = n;
  if (current % 3 == 0) {
    factors.add(3);
    current ~/= 3;
  }
  var maybePrime = 5;
  while (maybePrime <= root) {
    if (isPrime(maybePrime) && current % maybePrime == 0) {
      factors.add(maybePrime);
      current ~/= maybePrime;
    }
    maybePrime += 2;
    if (isPrime(maybePrime) && current % maybePrime == 0) {
      factors.add(maybePrime);
      current ~/= maybePrime;
    }
    maybePrime += 4;
  }
  return factors;
}

void main() {
  const n = 600851475143;
  var factors = solution(n);
  print(factors);
  print(factors.last);
}